src/components/TheHeader.vue

```diff
@@ -2,15 +2,27 @@
 import LogoImage from "@/assets/images/logo.svg?component";
 import HeaderActions from "@/components/header/HeaderActions.vue";
 import HeaderLinks from "@/components/header/HeaderLinks.vue";
+import HeaderBanner from "@/components/header/HeaderBanner.vue";
+import { computed } from "vue";
+import { useRoute } from "vue-router";
+const route = useRoute();
+
+const showActions = computed(() => {
+  return route.matched.some((el) => el.path !== "/login");
+});
+const showBanner = computed(() => {
+  return route.matched.some((el) => el.path === "/login");
+});
 </script>
 
 <template>
   <header class="header">
     <RouterLink :to="{ name: 'HomePage' }" class="logo">
       <LogoImage />
     </RouterLink>
-    <HeaderLinks />
-    <HeaderActions />
+    <HeaderLinks v-if="showActions" />
+    <HeaderActions v-if="showActions" />
+    <HeaderBanner v-if="showBanner" />
```

Comment:
Or instead of negating it (if you want to show banner only on one page 
```suggestion
   <template v-if="showActions">
      <HeaderLinks  />
      <HeaderActions  />
    </template>
    <HeaderBanner v-else="showBanner" />
```

Which seems to me to be a cleaner solution :)

Reply:
Over all thank you for kind wards and detailed and insightful review! I am going take your hints going forward. Have a great weekend.